catalog/db_branch_scanner.go

```diff
@@ -35,9 +42,36 @@ func NewDBBranchScanner(tx db.Tx, branchID int64, commitID CommitID, opts *DBSca
 		s.opts.BufferSize = DBScannerDefaultBufferSize
 	}
 	s.buf = make([]*DBScannerEntry, 0, s.opts.BufferSize)
+	commitsWhere, _ := getRelevantCommitsCondition(tx, branchID, commitID)
+	s.commitsWhere = commitsWhere
 	return s
 }
 
+func getRelevantCommitsCondition(tx db.Tx, branchID int64, commitID CommitID) (string, error) {
+	var maxCommitId CommitID
+	var commits []string
+	var commitsWhere string
+	if commitID == UncommittedID || commitID == CommittedID {
+		maxCommitId = MaxCommitID
+	} else {
+		maxCommitId = commitID
+	}
+	sql := "SELECT commit_id::text FROM catalog_commits WHERE branch_id = $1 AND commit_id <= $2 ORDER BY commit_id"
+	err := tx.Select(&commits, sql, branchID, maxCommitId)
+	if err != nil {
+		panic(err)
```

Comment:
Please don't panic on a DB failure, it might even just be transient.  Just fail the transaction, other things might continue to work,

Reply:
Is this still relevant?